### PR TITLE
Sequential parallel branches appear incorrectly while build in-progress

### PR DESCRIPTION
**Pull request by sophistifunk**
# Description

See [JENKINS-53816](https://issues.jenkins-ci.org/browse/JENKINS-53816).

# Submitter checklist
- [X] Link to JIRA ticket in description, if appropriate.
- [X] Change is code complete and matches issue description
- [X] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [X] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by sophistifunk** [comment]
Example pipeline that can be used to verify before/after changes: https://gist.github.com/sophistifunk/7b75a86658b80e82715d213820b673c7

**Comment by sophistifunk** [comment]
This doesn't address the shortcomings in the front-end code that are mainly responsible for JENKINS-55605 but some occurences of that problem will likely go away due to this fix.

**Comment by sophistifunk** [comment]
Build 5 was green, but I renamed some stuff so I'll try and get another green before I merge. You know, and a review.

File: blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeTest.java [hunk 1]
```diff
@@ -2462,6 +2402,78 @@ public void parallelStagesGroupsAndNestedStages() throws Exception {
...
+                                                  .collect(Collectors.joining(", "));
+
+            assertEquals("running node names", completeNodeNames, runningNodeNames);
+
+            loopCount++;
+
+        } while (run2.isBuilding());
+
+        // Sanity check, make sure we're *actually* checking stuff.
+        assertTrue("Checked multiple times while building", loopCount > 5);
```
Comment (olamy) [inline blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeTest.java]: I wonder if maybe we could have lower numbers on very slow/not powerful machines?

Comment (sophistifunk) [reply blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeTest.java]: I don't think it's a good idea. What use is a green test that's not really testing anything. I'd like to be able to use some other mechanism that time delays in the pipeline, something deterministic, but I don't know what that could really be. FWIW, the count running on my machine was > 90, so 5 seems pretty low.

**Pull request merged**